gensim/models/wrappers/ldamallet.py

```diff
@@ -244,3 +228,27 @@ def show_topic(self, topicid, topn=10):
 
     def print_topic(self, topicid, topn=10):
         return ' + '.join(['%.3f*%s' % v for v in self.show_topic(topicid, topn)])
+
+
+    def read_doctopics(self, fname, eps=1e-6):
+        """
+        Yield document topic vectors from MALLET's "doc-topics" format, as sparse gensim vectors.
+
+        """
+        fin = utils.smart_open(fname)
+        first_line = next(fin)
+        fin.close()
+        with utils.smart_open(fname) as fin:
+            if first_line.startswith("#doc "):  # check the file header
+                next(fin)   # skip the header line
+            for lineno, line in enumerate(fin):
+                parts = line.split()[2:]  # skip "doc" and "source" columns
+                if len(parts) == 2 * self.num_topics:
+                    doc = [(int(id), float(weight)) for id, weight in zip(parts[::2], parts[1::2]) if abs(float(weight)) > eps]
+                elif len(parts) == self.num_topics:
+                    doc = [(int(id), float(weight)) for id, weight in list(enumerate(parts)) if abs(float(weight)) > eps]
```

Comment:
very clever solution! :smile_cat: 

I think it would be helpful to add a comment over denoting why there's a need for the two different if-statements (e.g., this is for mallet version x, which formats the lines as `...`) just to make it clear what changes happened between the two versions.

For this particular step, there shouldn't be a need to use `list(enumerate(parts))`. Things should be fine with it being a plain `enumerate(parts)` since we are iterating over it with a loop anyway. Plus, this will save a lil memory not indirectly duplicating the parts list itself